### PR TITLE
dai: add warning when we skip .copy

### PR DESCRIPTION
**Pull request by mrajwa**
This patch logs warning message when we skip dai .copy() method.
Also the condition has been changed, there is no need to compare
positions - if there is nothing to copy we should terminate this
procedure regardless of positions.

Signed-off-by: Marcin Rajwa <marcin.rajwa@linux.intel.com>

### CODE REVIEW

File: src/audio/dai.c [hunk 1]
```diff
@@ -837,8 +837,9 @@ static int dai_copy(struct comp_dev *dev)
 		 dev->direction, copy_bytes,
 		 samples / buf->stream.channels);
 
-	/* return if it's not stream start */
-	if (!copy_bytes && dd->start_position != dev->position)
+	/* return if nothing to copy */
+	if (!copy_bytes)
+		comp_warn(dev, "dai_copy(): nothing to copy");
 		return 0;
```
Comment (ktrzcinx) [inline src/audio/dai.c:844]: There's {} missing

Comment (mrajwa) [reply src/audio/dai.c]: yes, thanks!

File: src/audio/dai.c [hunk 2]
```diff
@@ -837,9 +837,11 @@ static int dai_copy(struct comp_dev *dev)
 		 dev->direction, copy_bytes,
 		 samples / buf->stream.channels);
 
-	/* return if it's not stream start */
-	if (!copy_bytes && dd->start_position != dev->position)
+	/* return if nothing to copy */
+	if (!copy_bytes) {
+		comp_warn(dev, "dai_copy(): nothing to copy");
 		return 0;
+	}
```
Comment (dbaluta) [inline src/audio/dai.c:845]: Care to elaborate on why this change is needed. I mean, is there a specific usecase where you see an impact without this change?

I want to better understand the issue and also to have it written in the commit message. I guess, this is related to codec feature.

Comment (mrajwa) [reply src/audio/dai.c:845]: it's just to record this information about missing dai_copy in logs.

**Review by dbaluta — APPROVED** [review]

File: src/audio/dai.c (hunk 2 above)
Comment (dbaluta) [reply src/audio/dai.c:845]: @mrajwa ok, i see. thanks!

**Review by lgirdwood — APPROVED** [review]

**Review by paulstelian97 — APPROVED** [review]

**Comment by lgirdwood** [comment]
Jenkins CI BAT failure on BYT sounds fine, could be script/DUT issue.
